src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -253,18 +252,20 @@ void GcsServer::InitGcsActorManager(const GcsInitData &gcs_init_data) {
 
   if (RayConfig::instance().gcs_actor_scheduling_enabled()) {
     RAY_CHECK(gcs_resource_manager_ && gcs_resource_scheduler_);
-    scheduler = std::make_unique<GcsBasedActorScheduler>(
+    gcs_actor_scheduler_ = std::make_unique<GcsBasedActorScheduler>(
         main_service_, gcs_table_storage_->ActorTable(), *gcs_node_manager_, gcs_pub_sub_,
         gcs_resource_manager_, gcs_resource_scheduler_, schedule_failure_handler,
         schedule_success_handler, raylet_client_pool_, client_factory);
   } else {
-    scheduler = std::make_unique<RayletBasedActorScheduler>(
+    gcs_actor_scheduler_ = std::make_unique<RayletBasedActorScheduler>(
         main_service_, gcs_table_storage_->ActorTable(), *gcs_node_manager_, gcs_pub_sub_,
         schedule_failure_handler, schedule_success_handler, raylet_client_pool_,
         client_factory);
   }
   gcs_actor_manager_ = std::make_shared<GcsActorManager>(
-      std::move(scheduler), gcs_table_storage_, gcs_pub_sub_, *runtime_env_manager_,
+      static_cast<std::shared_ptr<GcsActorSchedulerInterface>>(
```

Comment:
We probably need to make GcsActorManager accept raw of ref of the object instead of shared ptr.
The updates here in this PR move the ownership of gcs_actor_scheduler_ from gcs actor manager to gcs_server.